### PR TITLE
use jib to build a memory product docker image with maven

### PR DESCRIPTION
**Pull request by mbaechler**
With this PR, running `mvn package` produces a docker image of the memory product and push it in your local daemon.

It can be disabled (for env without docker) with `-Djib.skip`.

The cli and glowroot are both working as expected.

The jdk image is the default one: gcr.io/distroless/java:8 , it means no shell, nothing else than java.

Finally, the docker image takes 217MiB:
* 69MiB are java dependencies
* glowroot takes 17MiB
* james code takes 6MiB
* the base image takes 125MiB

Let me know what you think about it

### CODE REVIEW

File: server/container/guice/memory-guice/pom.xml [hunk 1]
```diff
@@ -225,6 +237,50 @@
...
+                                    <jvmFlag>-Dworking.directory=/root</jvmFlag>
+                                    <jvmFlag>-Dlogback.configurationFile=/root/conf/logback.xml</jvmFlag>
+                                </jvmFlags>
+                                <ports>
+                                    <port>25</port>
+                                    <port>80</port>
+                                    <port>143</port>
+                                    <port>465</port>
+                                    <port>587</port>
+                                    <port>993</port>
```
Comment (remk) [inline server/container/guice/memory-guice/pom.xml]: We don't expose port `4000` for glowroot ?

Comment (mbaechler) [reply server/container/guice/memory-guice/pom.xml]: As it's not started by default, I did not. But if you have another opinion, let me know.

Comment (remk) [reply server/container/guice/memory-guice/pom.xml]: Ok, it's good for me then.

**Comment by rouazana** [comment]
(please add a JIRA)

**Comment by mbaechler** [comment]
superseded by https://github.com/linagora/james-project/pull/3018

**Pull request closed without merging**